lib/provisioner/plugin/utils.rb

```diff
@@ -49,6 +51,21 @@ def to_json(*a)
         end
       end
 
+      # Decodes a base64 string into a file
+      def decode_string_to_file(string, outfile, mode = 0o600)
+        ::FileUtils.mkdir_p(::File.dirname(outfile))
+        ::File.open(outfile, 'wb', mode) { |f| f.write(::Base64.decode64(string)) }
+      end
+
+      # Writes out an SSH file, given a path and task JSON, returns fully-qualified path to file
+      def write_ssh_file(path, task)
```

Comment:
suggest adding a ``return nil if task['config']['provider']['provisioner']['ssh_keyfile'].nil?`` instead of the 3 checks we do below.  

Note this changes the return value of the function in this case though.  see other comment